### ISSUE TITLE
TypeError

### ISSUE THREAD

**Issue by MrLanceken** [upchuk/commerce_bpost#3]
It happens at:
checkout/1/order_information
I don't have much info.
I can give you full stack trace.

`TypeError: Argument 2 passed to Drupal\Component\Plugin\PluginManagerBase::createInstance() must be of the type array, null given, called in /modules/contrib/commerce_bpost/src/Plugin/Commerce/ShippingMethod/Bpost.php on line 281 in Drupal\Component\Plugin\PluginManagerBase->createInstance() (line 71 of core/lib/Drupal/Component/Plugin/PluginManagerBase.php).`

Full stack:

`Drupal\Component\Plugin\PluginManagerBase->createInstance('default', NULL) (Line: 281)
Drupal\commerce_bpost\Plugin\Commerce\ShippingMethod\Bpost->calculateRates(Object) (Line: 89)
Drupal\commerce_shipping\ShipmentManager->calculateRates(Object) (Line: 92)
Drupal\commerce_shipping\Plugin\Field\FieldWidget\ShippingRateWidget->formElement(Object, 0, Array, Array, Object) (Line: 344)
Drupal\Core\Field\WidgetBase->formSingleElement(Object, 0, Array, Array, Object) (Line: 211)
Drupal\Core\Field\WidgetBase->formMultipleElements(Object, Array, Object) (Line: 113)
Drupal\Core\Field\WidgetBase->form(Object, Array, Object) (Line: 178)
Drupal\Core\Entity\Entity\EntityFormDisplay->buildForm(Object, Array, Object) (Line: 273)
Drupal\commerce_shipping\Plugin\Commerce\CheckoutPane\ShippingInformation->buildPaneForm(Array, Object, Array) (Line: 542)
Drupal\commerce_checkout\Plugin\Commerce\CheckoutFlow\CheckoutFlowWithPanesBase->buildForm(Array, Object, 'order_information')
call_user_func_array(Array, Array) (Line: 532)
Drupal\Core\Form\FormBuilder->retrieveForm('commerce_checkout_flow_multistep_default', Object) (Line: 278)
Drupal\Core\Form\FormBuilder->buildForm(Object, Object) (Line: 219)
Drupal\Core\Form\FormBuilder->getForm(Object, 'order_information') (Line: 143)
Drupal\commerce_checkout\Controller\CheckoutController->formPage(Object)
call_user_func_array(Array, Array) (Line: 123)
Drupal\Core\EventSubscriber\EarlyRenderingControllerWrapperSubscriber->Drupal\Core\EventSubscriber\{closure}() (Line: 573)
Drupal\Core\Render\Renderer->executeInRenderContext(Object, Object) (Line: 124)
Drupal\Core\EventSubscriber\EarlyRenderingControllerWrapperSubscriber->wrapControllerExecutionInRenderContext(Array, Array) (Line: 97)
Drupal\Core\EventSubscriber\EarlyRenderingControllerWrapperSubscriber->Drupal\Core\EventSubscriber\{closure}() (Line: 151)
Symfony\Component\HttpKernel\HttpKernel->handleRaw(Object, 1) (Line: 68)
Symfony\Component\HttpKernel\HttpKernel->handle(Object, 1, 1) (Line: 57)
Drupal\Core\StackMiddleware\Session->handle(Object, 1, 1) (Line: 47)
Drupal\Core\StackMiddleware\KernelPreHandle->handle(Object, 1, 1) (Line: 106)
Drupal\page_cache\StackMiddleware\PageCache->pass(Object, 1, 1) (Line: 85)
Drupal\page_cache\StackMiddleware\PageCache->handle(Object, 1, 1) (Line: 47)
Drupal\Core\StackMiddleware\ReverseProxyMiddleware->handle(Object, 1, 1) (Line: 52)
Drupal\Core\StackMiddleware\NegotiationMiddleware->handle(Object, 1, 1) (Line: 23)
Stack\StackedHttpKernel->handle(Object, 1, 1) (Line: 708)
Drupal\Core\DrupalKernel->handle(Object) (Line: 19)`

Hope this helps.

**Comment by upchuk** [comment]
This tells me nothing and I will close it. Since you are using this module for free and you see a willingness from me to fix and help out, please be kind enough to provide me with decent information as to what is going wrong, how can it be reproduced, so I can fix it.

**Comment by MrLanceken** [comment]
> This tells me nothing and I will close it. Since you are using this module for free and you see a willingness from me to fix and help out, please be kind enough to provide me with decent information as to what is going wrong, how can it be reproduced, so I can fix it.

Hi,
I will help, but I don't have much more info. What do you need?

**Comment by upchuk** [comment]
Im not sure what is wrong there. Did you configure the shipping method to use Home Delivery at least?

**Comment by MrLanceken** [comment]
There is nothing to configure at /admin/commerce/shipping-methods/manage/3  about 'home delivery'.
Only:
plugin (radio button - bPost)
API settings (did not add them yet)
Rate amounts

I did not activate the bPost Pickup Point plugin.

**Comment by upchuk** [comment]
![Screen Shot 2021-03-26 at 17 24 36](https://user-images.githubusercontent.com/5848933/112654554-cd213c00-8e4f-11eb-87a4-6c14358dad8a.png)

Take a screenshot of this page please

**Comment by MrLanceken** [comment]
![image](https://user-images.githubusercontent.com/81426502/112663883-9ea85e80-8e59-11eb-97d7-f2ee398a3322.png)

**Comment by upchuk** [comment]
Did you create a store which also has shipping countries, including BE?

**Comment by MrLanceken** [comment]
Yes I've selected Belgium and Netherlands here: /admin/commerce/config/stores under 'supported shipping countries'

**Comment by MrLanceken** [comment]
I also found this error. But updating 'updatedb' with drush did not remove the error ...

![image](https://user-images.githubusercontent.com/81426502/112833915-a5bcb000-9097-11eb-9c55-890b1df8fce3.png)
